datafusion/src/execution/context.rs

```diff
@@ -710,17 +710,17 @@ impl Default for ExecutionConfig {
             optimizers: vec![
                 Arc::new(ConstantFolding::new()),
                 Arc::new(EliminateLimit::new()),
-                Arc::new(AggregateStatistics::new()),
                 Arc::new(ProjectionPushDown::new()),
                 Arc::new(FilterPushDown::new()),
                 Arc::new(SimplifyExpressions::new()),
-                Arc::new(HashBuildProbeOrder::new()),
                 Arc::new(LimitPushDown::new()),
             ],
             physical_optimizers: vec![
                 Arc::new(CoalesceBatches::new()),
                 Arc::new(Repartition::new()),
                 Arc::new(AddCoalescePartitionsExec::new()),
+                Arc::new(AggregateStatistics::new()),
```

Comment:
I think  we probably want to run the `AggregateStatistics` passes first (prior to running the other optimizers); Even though it probably doesn't matter at this time, if any of those optimizers were to use the statistics now or in the future it might lead to tricky bugs.

Reply:
We may want to run all the optimizers multiple rounds until no further plan tree changing or we reached an upper limit round of optimization, in order to keep stats up to date as well as optimization consistency.

Reply:
Agree with you both. For now I am simply promoting it to run first.